src/priority/priority.py

```diff
@@ -295,6 +295,11 @@ def __init__(self, maximum_streams=1000):
         self._root_stream = Stream(stream_id=0, weight=1)
         self._root_stream.active = False
         self._streams = {0: self._root_stream}
+
+        if not isinstance(maximum_streams, int):
+            raise TypeError("maximum_streams must be an int.")
+        if maximum_streams <= 0:
+            raise TypeError("maximum_streams must be a positive integer.")
```

Comment:
Soooo...*strictly* this should be a `ValueError`, not a `TypeError`, as the type of the input is acceptable, but it is out of range.